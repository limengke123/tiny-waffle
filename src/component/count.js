import React from 'react';
import PropTypes from 'prop-types';

class Count extends React.Component {
  static propTypes = {
    count: PropTypes.number
  };

  render() {
    const { count } = this.props;
    return <div>{count}</div>;
  }
}

export { Count };
