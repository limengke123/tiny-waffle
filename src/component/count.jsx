import React from 'react'
import PropTypes from 'prop-types'

class Count extends React.Component {
    static propTypes = {
        count: PropTypes.number
    }

    static defaultProps = {
        count: 0
    }

    render() {
        const { count } = this.props
        return <div>{count}</div>
    }
}

export { Count }
