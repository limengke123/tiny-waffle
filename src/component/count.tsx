import React from 'react'
import styles from '../style/component/count.module.scss'
import { CounterState } from '../page/counter'

export interface CountProps {
    countInfo: CounterState['count']
}

export class Count extends React.PureComponent<CountProps> {
    render() {
        const { countInfo } = this.props
        const { number, modifyInfo } = countInfo
        const { time, type, step } = modifyInfo
        return (
            <div>
                <div className={styles.count}>
                    number:
                    {number}
                </div>
                <div className={styles['sub-info']}>
                    time:
                    {time}
                </div>
                <div className={styles['sub-info']}>
                    type:
                    {type}
                </div>
                <div className={styles['sub-info']}>
                    step:
                    {step}
                </div>
            </div>
        )
    }
}
