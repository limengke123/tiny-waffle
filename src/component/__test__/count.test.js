import React from 'react'
import { render } from 'enzyme'
import { Count } from '../count'

describe('count', () => {
    it('should render correctly', () => {
        const wrapper = render(<Count />)
        expect(wrapper).toMatchSnapshot()

        const countInfo = {
            number: 5,
            modifyInfo: {
                time: '2019-2-12 12:24:39',
                type: 'Addition',
                step: 2
            }
        }
        const wrapper1 = render(<Count countInfo={countInfo} />)
        expect(wrapper1).toMatchSnapshot()
    })
})
