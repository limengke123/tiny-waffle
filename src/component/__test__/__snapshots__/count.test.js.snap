// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`count should render correctly 1`] = `
<div>
  <div
    class="count"
  >
    number:0
  </div>
  <div
    class="sub-info"
  >
    time:
  </div>
  <div
    class="sub-info"
  >
    type:
  </div>
  <div
    class="sub-info"
  >
    step:1
  </div>
</div>
`;

exports[`count should render correctly 2`] = `
<div>
  <div
    class="count"
  >
    number:5
  </div>
  <div
    class="sub-info"
  >
    time:2019-2-12 12:24:39
  </div>
  <div
    class="sub-info"
  >
    type:Addition
  </div>
  <div
    class="sub-info"
  >
    step:2
  </div>
</div>
`;
